src/main/java/ch/njol/skript/sections/SecLoop.java

```diff
@@ -131,19 +123,24 @@ protected TriggerItem walk(Event event) {
 		Iterator<?> iter = currentIter.get(event);
 		if (iter == null) {
 			iter = expr instanceof Variable<?> variable ? variable.variablesIterator(event) : expr.iterator(event);
-			if (iter != null) {
-				if (iter.hasNext())
-					currentIter.put(event, iter);
-				else
-					iter = null;
+			if (iter != null && iter.hasNext()) {
+				currentIter.put(event, iter);
+				next.put(event, iter.next());
 			}
 		}
-		if (iter == null || !iter.hasNext()) {
+
+		if (iter == null || next.get(event) == null) {
 			exit(event);
 			debug(event, false);
 			return actualNext;
 		} else {
-			current.put(event, iter.next());
+			previous.put(event, current.get(event));
+			current.put(event, next.get(event));
+			if (iter.hasNext()) {
+				next.put(event, iter.next());
+			} else {
+				next.put(event, null);
+			}
```

Comment:
So what exactly is it that you're asking me to do here? I understand you're stating the potential inconsistency of the values depending on how they are handled. However, personally, I don't see how this would create any **new** issues as it's handling the values the exact same way. Wouldn't it also be safe to assume that currently with only the current `loop-value` that the values could be altered before the user accesses it? or even values that are being altered before and/or after they are the current value? I'm failing to understand the "main issue" persay, since the only "big" change is where the value is being stored to be accessed from. Maybe it's just my lack of Java knowledge.
If you could explain more on how these changes would cause these potential inconsistencies but the current system doesn't and also what it is you're looking for me to do/change as this review may explicitly say "requested changes" wasn't approved neither.